### PR TITLE
Create rule "module-folder-structure"

### PR DESCRIPTION
**Pull request by sergioflores-j**
This PR adds the rule `module-folder-structure` to the project.
Solves #2.

Also, a small refactor in `src/rules/registry.ts` was made.

## TODOs:

- [X] - Create rule boilerplate
- [X] Register rule
- [X] Create test cases
- [X] Implement basic patterns check
- [X] Implement strict mode
- [x] Implement glob patterns check
- [X] Create example docs - readme
- [x] Create docs - docExampleConfigs

### CODE REVIEW

**Comment by flaviostutz** [comment]
@sergioflores-j please fix the conflicts and the pipeline run results so we can review this one next :)

**Comment by sergioflores-j** [comment]
@flaviostutz PR should be ready now! 

Can you review it?

File: src/rules/module-folder-structure.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      },
+      {
+        description: 'Loosely requires module structure. The module should contain **at least** this set of folders, but can still have others.',
+        config: {
+          strict: false,
+          folders: ['src', 'docs', 'libs'],
+        },
+      },
+      {
+        description: 'Strictly requires module structure. No extra folders allowed, should match exactly.',
```
Comment (flaviostutz) [inline src/rules/module-folder-structure.ts:130]: Writing style discussion: the text is very well written, but maybe you could describe it more as a concrete "example" of what it's doing ("documentation by examples"), like:

"Allow only folder 'src/test' or any folders inside 'src' that has a name 'util'".

We've been writing like this in other rules. Maybe a combination of both writing styles can help people understand. What do you think?

**Review by flaviostutz — APPROVED** [review]
Nice!!! I had some small comments only about writing style that I would like to hear from you.

File: src/rules/module-folder-structure.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            rule: rule.name,
+            module,
+          });
+        }
+      }
+
+      if (strict) {
+        const foldersMatchingAllPatterns = fg.sync(requiredFolderPatterns, fgConfig);
+
+        // ? If there's no match for all patterns, then we shouldn't check extra folders
```
[the remark this answers is not in the document]
Comment (sergioflores-j) [reply src/rules/module-folder-structure.ts:81]: It's a vscode extension hahaha

https://marketplace.visualstudio.com/items?itemName=aaron-bond.better-comments

File: src/rules/module-folder-structure.ts (hunk 1 above)
Comment (sergioflores-j) [reply src/rules/module-folder-structure.ts:130]: I agree! I was mainly inspired by the writing style from the `required-files` rule. Maybe we could revisit them all at a later stage.

**Pull request merged**